ui-grid/ui-grid.d.ts

```diff
@@ -2324,6 +2330,15 @@ declare module uiGrid {
         export interface paginationChangedHandler {
             (newPage: number, pageSize: number): void;
         }
+
+        export interface sortChangedHandler {
+            /**
+             * The callback for sortChanged
+             * @param {uiGrid.IGridInstance} grid the grid instance
+             * @param {Array<uiGrid.IGridColumn>} sortColumns sorting columns
+             */
+            (grid: uiGrid.IGridInstance, sortColumns: Array<uiGrid.IGridColumn>): void;
```

Comment:
I know this PR was opened before #5972 was merged, but I think we should use the generic versions of `IGridInstance` and `IGridColumn` here to be consistent and provide type helps when using this feature.
In order to do this, change: 
`uiGrid.IGridInstance` to `uiGrid.IGridInstanceOf<TEntity>`,
`uiGrid.IGridColumn` to `uiGrid.IGridColumnOf<TEntity>`,
`sortChangedHandler` to `sortChangedHandler<TEntity>`,
then add generics up the chain as necessary.